packages/expo-updates/ios/EXUpdates/AppLoader/FileDownloader.swift

```diff
@@ -832,7 +854,16 @@ public final class FileDownloader {
 
       successBlock(data, response)
     }
-    task.resume()
+
+    if let progressBlock = progressBlock, let task = task {
+      progressObservationsLock.lock()
```

Comment:
btw in this case
maybe use a serial DispatchQueue.async so we don't need lock and make sure thread safety
it may have a little performance gain

Reply:
shared the reason [here](https://github.com/expo/expo/pull/38307#discussion_r2233062630) for not doing async for progress updates. maybe we share a queue for all state machine events? cc - @douglowder need some help 🙏